### ISSUE TITLE
range cannot start with a negative number

### ISSUE THREAD

**Issue by fj0r** [nushell/nushell#9177]
### Describe the bug

Start with a negative number, it will first parse `-1` into flag` surround it with quotes, it will be parsed as a string.

### How to reproduce

```
'abc' | str substring -1..                                                                                                                   
Error: nu::parser::unknown_flag

  × The `str substring` command doesn't have flag `-1`.
   ╭─[entry #18:1:1]
 1 │ 'abc' | str substring -1..
   ·                        ┬
   ·                        ╰── unknown flag
   ╰────
  help: Available flags: --help(-h), --grapheme-clusters(-g), --utf-8-bytes(-b). Use `--help` for more information.
```

### Expected behavior

`'abc' | str substring -1.. ` should get `c`

### Configuration

| key                | value                                                                  
                                                                                         |
| ------------------ | -----------------------------------------------------------------------
---------------------------------------------------------------------------------------- |
| version            | 0.79.0                                                                 
                                                                                         |
| branch             |                                                                        
                                                                                         |
| commit_hash        | a1b72611215dbfca257351003204a80c83859e05                               
                                                                                         |
| build_os           | linux-x86_64                                                           
                                                                                         |
| build_target       | x86_64-unknown-linux-musl                                              
                                                                                         |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)                                    
                                                                                         |
| rust_channel       | 1.66.1-x86_64-unknown-linux-gnu                                        
                                                                                         |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)                                    
                                                                                         |
| build_time         | 2023-04-25 20:27:44 +00:00                                             
                                                                                         |
| build_rust_channel | release                                                                
                                                                                         |
| features           | default, zip                                                           
                                                                                         |
| installed_plugins  | custom-value generate, custom-value generate2, custom-value update, fro
m eml, from ics, from ini, from vcf, gstat, inc, query, query json, query web, query xml |

**Comment by amtoine** [comment]
yeah, the parse sees that as the start of a short option :open_mouth: 
does `'abc' | str substring (-1..)` work for you on this? :relieved:

**Comment by fj0r** [comment]
it works, but it's not very intuitive.

**Comment by NotTheDr01ds** [comment]
Also:

```nushell
'abc' | str substring (-1)..
# => c
```

I feel like the behavior here is correct, but perhaps the error message could be improved.
